src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -184,23 +185,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             Instant.now());
 
     messageStore.addMessage(message);
-    UUID mentionedUser = notificationStore.getuserMentioned(message.getContent());
-    //Add notification if a user is mentioned
-      if(mentionedUser != null) {
-        String notificationinfo = String.format("%s mentioned you in <a href=\"/chat/%s\">%s</a>: %s", username, conversationTitle,conversationTitle, message.getContent());
-        Notification notification =
-                new Notification(UUID.randomUUID(),
-                                 user.getId(),
-                                 conversation.getId(),
-                                 notificationinfo,
-                                 mentionedUser);
-        notificationStore.addNotification(notification);
-      }
-
-      //Log Activity for message creation
-      String activityDescription = String.format("%s sent a message to %s: %s", username, conversationTitle, messageContent);
-      Activity activity = new Activity(UUID.randomUUID(), activityDescription, Instant.now(), ActivityTypeEnum.MESSAGE_ADDED);
-      activityStore.addActivity(activity);
+    notificationService.generateMentionNotification(message, conversation, user);
```

Comment:
this is the only actual line change in this green block; the rest is just indentation